### ISSUE TITLE
WPInternals crashing/freezing when pressing "unlock" button(RM-1045, Spec B)

### ISSUE THREAD

**Issue by MackaG** [ReneLergner/WPinternals#69]
### Description

I already had this issue on 640 lte while trying to use EDE file, when using EDP everything was fine, until I bought 930 and its simply crashing

### Steps To Reproduce

Connect lumia 930 in flash mode to WPinternals
Download latest firmware, ede and edp files
Download firmware for RM-1085 with compatibile OS version
Select corresponding files
Click "Unlock"

### Expected behavior

WPInternals freezes and closes

### WPinternals Version

2.9.2 and 2.3

### Windows Version

Windows 11 23h2 and Windows 7

### Phone model

Lumia 930 and 640 LTE

### Relevant Assets

I'm on different PC, but I noticed error like: "Indeks wykracza poza granice tablicy" which means Index oudside bounds of array

**Comment by MackaG** [comment]
Important part of log:
Adding emergency files to repository: C:\ede.ede
Type: RM-1045
Error: Indeks wykracza poza granice tablicy

**Comment by gus33000** [comment]
Hi, can you please attach the log file? Thank you

**Comment by MackaG** [comment]
Ok


2024-05-05 23:01:28.708: Windows Phone Internals version 2.3.6584.1869
2024-05-05 23:01:28.710: Copyright Heathcliff74 / wpinternals.net
2024-05-05 23:01:28.919: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-05 23:01:28.919: Device path: \\?\usb#vid_0421&pid_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-05 23:01:28.919: Connected device: Lumia
2024-05-05 23:01:28.919: Mode: Flash
2024-05-05 23:01:29.060: Security flags: 0x00007BFF
2024-05-05 23:01:29.087: Platform Name: Nokia.MSM8974.P6180.2.2
2024-05-05 23:01:29.112: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-05 23:01:29.143: Root Key Hash: 80 0E EB 50 8F 7B BD F1 2A 19 26 26 21 FD 83 72 97 A3 B0 62 FE 2A 70 78 C0 F3 16 7E 57 F2 12 17
2024-05-05 23:01:29.143: Platform Secure Boot Status: True
2024-05-05 23:01:29.144: Uefi Secure Boot Status: True
2024-05-05 23:01:29.144: Effective Secure Boot Status: True
2024-05-05 23:01:29.145: Bootloader Security Qfuse Status: True
2024-05-05 23:01:29.145: Bootloader Security Authentication Status: False
2024-05-05 23:01:29.147: Bootloader Security Rdc Status: False
2024-05-05 23:01:29.147: Effective Bootloader Security Status: True
2024-05-05 23:01:29.147: Native Debug Status: False
2024-05-05 23:01:29.418: Phone type: RM-1045
2024-05-05 23:01:29.418: Product code: 059W0X6
2024-05-05 23:01:29.418: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-05 23:01:29.418: Firmware version: 02540.00019.15236.45000
2024-05-05 23:01:29.418: IMEI: 353633062332557
2024-05-05 23:01:29.419: Flash app: 2.59
2024-05-05 23:01:29.419: Flash protocol: 2.34
2024-05-05 23:01:29.419: SecureBoot: Enabled
2024-05-05 23:01:29.419: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-05 23:01:29.419: JTAG: Disabled
2024-05-05 23:01:29.419: Bootloader: Lumia Bootloader Spec B
2024-05-05 23:01:29.419: ProductCode: 059W0X6
2024-05-05 23:01:29.420: ProductType: RM-1045
2024-05-05 23:01:59.953: Adding FFU to repository: C:\ffu1.ffu
2024-05-05 23:01:59.953: Platform ID: Nokia.MSM8974.P6180
2024-05-05 23:01:59.953: Firmware version: 02540.00019.15236.45000
2024-05-05 23:01:59.953: OS version: 6.3.9651.2
2024-05-05 23:02:00.003: Adding emergency files to repository: C:\edp.edp
2024-05-05 23:02:00.003: Type: RM-1045
2024-05-05 23:02:00.014: Error: Indeks wykracza poza granice tablicy.
2024-05-05 23:02:09.229: Windows Phone Internals version 2.3.6584.1869
2024-05-05 23:02:09.230: Copyright Heathcliff74 / wpinternals.net
2024-05-05 23:02:09.439: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-05 23:02:09.439: Device path: \\?\usb#vid_0421&pid_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-05 23:02:09.439: Connected device: Lumia
2024-05-05 23:02:09.439: Mode: Flash
2024-05-05 23:02:09.557: Security flags: 0x00007BFF
2024-05-05 23:02:09.581: Platform Name: Nokia.MSM8974.P6180.2.2
2024-05-05 23:02:09.605: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-05 23:02:09.626: Root Key Hash: 80 0E EB 50 8F 7B BD F1 2A 19 26 26 21 FD 83 72 97 A3 B0 62 FE 2A 70 78 C0 F3 16 7E 57 F2 12 17
2024-05-05 23:02:09.627: Platform Secure Boot Status: True
2024-05-05 23:02:09.627: Uefi Secure Boot Status: True
2024-05-05 23:02:09.628: Effective Secure Boot Status: True
2024-05-05 23:02:09.628: Bootloader Security Qfuse Status: True
2024-05-05 23:02:09.628: Bootloader Security Authentication Status: False
2024-05-05 23:02:09.629: Bootloader Security Rdc Status: False
2024-05-05 23:02:09.629: Effective Bootloader Security Status: True
2024-05-05 23:02:09.629: Native Debug Status: False
2024-05-05 23:02:09.920: Phone type: RM-1045
2024-05-05 23:02:09.920: Product code: 059W0X6
2024-05-05 23:02:09.920: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-05 23:02:09.920: Firmware version: 02540.00019.15236.45000
2024-05-05 23:02:09.920: IMEI: 353633062332557
2024-05-05 23:02:09.920: Flash app: 2.59
2024-05-05 23:02:09.920: Flash protocol: 2.34
2024-05-05 23:02:09.920: SecureBoot: Enabled
2024-05-05 23:02:09.920: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-05 23:02:09.920: JTAG: Disabled
2024-05-05 23:02:09.921: Bootloader: Lumia Bootloader Spec B
2024-05-05 23:02:09.921: ProductCode: 059W0X6
2024-05-05 23:02:09.921: ProductType: RM-1045
06.05.2024 00:37:28.262: Found device on interface: 53f56307-b6bf-11d0-94f2-00a0c91efb8b
06.05.2024 00:37:28.265:   Device path: \\?\usbstor#disk&ven_dm&prod_easymaxtouch&rev_0.01#7&28a132c&0&59333931323515050b02&0#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}
06.05.2024 00:37:28.265: Found device on interface: 53f56307-b6bf-11d0-94f2-00a0c91efb8b
06.05.2024 00:37:28.265:   Device path: \\?\scsi#disk&ven_st500lm0&prod_00-1ej162#7&136fce66&0&000000#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}
06.05.2024 00:37:28.265: Found device on interface: 53f56307-b6bf-11d0-94f2-00a0c91efb8b
06.05.2024 00:37:28.265:   Device path: \\?\scsi#disk&ven_nvme&prod_pm991a_nvme_sams#5&1fb7b919&0&000000#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}
06.05.2024 00:37:28.275: Found device on interface: 86e0d1e0-8089-11d0-9ce4-08003e301f73
06.05.2024 00:37:28.275:   Device path: \\?\root#ports#0000#{86e0d1e0-8089-11d0-9ce4-08003e301f73}
06.05.2024 00:37:28.275: Found device on interface: 86e0d1e0-8089-11d0-9ce4-08003e301f73
06.05.2024 00:37:28.275:   Device path: \\?\root#ports#0001#{86e0d1e0-8089-11d0-9ce4-08003e301f73}
06.05.2024 00:39:12.888: Found device on interface: 08324f9c-b621-435c-859b-ae4652481b7c
06.05.2024 00:39:12.888:   Device path: \\?\USB#VID_0421&PID_06FC&MI_03#6&3aa93d4e&0&0003#{08324f9c-b621-435c-859b-ae4652481b7c}
06.05.2024 00:39:12.946: Found Lumia. Mode: Normal.
06.05.2024 00:39:14.328: Operator: 000-33
06.05.2024 00:39:14.334: Manufacturer Model Name: RM-1045_1006
06.05.2024 00:39:14.395: Product Code: 059W0X6
06.05.2024 00:39:14.398: Firmware: 02540.00019.15236.45005
06.05.2024 00:39:14.415: IMEI: 353633062332557
06.05.2024 00:39:14.447: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
06.05.2024 00:39:14.463: Bluetooth MAC: 10 2F 6B BD 91 12
06.05.2024 00:39:14.512: WLAN MAC: 10 2F 6B BF E5 68
06.05.2024 00:39:14.544: Bootloader Security: Enabled
06.05.2024 00:39:14.588: Simlock: Unlocked
06.05.2024 00:40:36.946: Start unlock. Phone needs to switch to Flash-mode
06.05.2024 00:40:39.94: Switching to Flash-mode
06.05.2024 00:40:39.102: Rebooting phone to Flash mode
06.05.2024 00:40:41.670: Phone disconnected.
06.05.2024 00:40:46.587: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
06.05.2024 00:40:46.587:   Device path: \\?\USB#VID_0421&PID_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
06.05.2024 00:40:46.646: Found Lumia. Mode: Flash.
06.05.2024 00:40:46.781: Effective Bootloader Security Status: True
06.05.2024 00:40:46.820: Security flags: 0x00007BFF
06.05.2024 00:40:46.843: Platform Name: Nokia.MSM8974.P6180.2.2
06.05.2024 00:40:46.864: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
06.05.2024 00:40:46.885: Root Key Hash: 80 0E EB 50 8F 7B BD F1 2A 19 26 26 21 FD 83 72 97 A3 B0 62 FE 2A 70 78 C0 F3 16 7E 57 F2 12 17
06.05.2024 00:40:46.886: Platform Secure Boot Status: True
06.05.2024 00:40:46.886: Uefi Secure Boot Status: True
06.05.2024 00:40:46.886: Effective Secure Boot Status: True
06.05.2024 00:40:46.887: Bootloader Security Qfuse Status: True
06.05.2024 00:40:46.887: Bootloader Security Authentication Status: False
06.05.2024 00:40:46.888: Bootloader Security Rdc Status: False
06.05.2024 00:40:46.888: Effective Bootloader Security Status: True
06.05.2024 00:40:46.888: Native Debug Status: False
2024-05-06 00:49:44.833: Windows Phone Internals version 2.8.6950.39616
2024-05-06 00:49:44.835: Copyright Heathcliff74 / wpinternals.net
2024-05-06 00:49:45.079: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 00:49:45.079: Device path: \\?\usb#vid_0421&pid_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 00:49:45.079: Connected device: Lumia
2024-05-06 00:49:45.079: Mode: Flash
2024-05-06 00:49:45.199: Security flags: 0x00007BFF
2024-05-06 00:49:45.220: Platform Name: Nokia.MSM8974.P6180.2.2
2024-05-06 00:49:45.244: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 00:49:45.266: Root Key Hash: 80 0E EB 50 8F 7B BD F1 2A 19 26 26 21 FD 83 72 97 A3 B0 62 FE 2A 70 78 C0 F3 16 7E 57 F2 12 17
2024-05-06 00:49:45.267: Platform Secure Boot Status: True
2024-05-06 00:49:45.269: Uefi Secure Boot Status: True
2024-05-06 00:49:45.269: Effective Secure Boot Status: True
2024-05-06 00:49:45.270: Bootloader Security Qfuse Status: True
2024-05-06 00:49:45.270: Bootloader Security Authentication Status: False
2024-05-06 00:49:45.270: Bootloader Security Rdc Status: False
2024-05-06 00:49:45.271: Effective Bootloader Security Status: True
2024-05-06 00:49:45.271: Native Debug Status: False
2024-05-06 00:49:45.547: Phone type: RM-1045
2024-05-06 00:49:45.547: Product code: 059W0X6
2024-05-06 00:49:45.547: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-06 00:49:45.547: Firmware version: 02540.00019.15236.45000
2024-05-06 00:49:45.547: IMEI: 353633062332557
2024-05-06 00:49:45.548: Flash app: 2.59
2024-05-06 00:49:45.548: Flash protocol: 2.34
2024-05-06 00:49:45.548: SecureBoot: Enabled
2024-05-06 00:49:45.548: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-06 00:49:45.548: JTAG: Disabled
2024-05-06 00:49:45.548: Bootloader: Lumia Bootloader Spec B
2024-05-06 00:49:45.548: ProductCode: 059W0X6
2024-05-06 00:49:45.549: ProductType: RM-1045
2024-05-06 00:50:21.731: FFU not added, because it was already present in the repository.
2024-05-06 00:50:21.733: Adding emergency files to repository: C:\edp.edp
2024-05-06 00:50:21.733: Type: RM-1045
2024-05-06 00:50:21.743: Error: Indeks wykracza poza granice tablicy.
2024-05-06 01:01:09.204: Windows Phone Internals version 2.8.6950.39616
2024-05-06 01:01:09.205: Copyright Heathcliff74 / wpinternals.net
2024-05-06 01:01:09.426: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 01:01:09.426: Device path: \\?\usb#vid_0421&pid_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 01:01:09.426: Connected device: Lumia
2024-05-06 01:01:09.426: Mode: Flash
2024-05-06 01:01:09.544: Security flags: 0x00007BFF
2024-05-06 01:01:09.566: Platform Name: Nokia.MSM8974.P6180.2.2
2024-05-06 01:01:09.590: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 01:01:09.614: Root Key Hash: 80 0E EB 50 8F 7B BD F1 2A 19 26 26 21 FD 83 72 97 A3 B0 62 FE 2A 70 78 C0 F3 16 7E 57 F2 12 17
2024-05-06 01:01:09.614: Platform Secure Boot Status: True
2024-05-06 01:01:09.614: Uefi Secure Boot Status: True
2024-05-06 01:01:09.615: Effective Secure Boot Status: True
2024-05-06 01:01:09.616: Bootloader Security Qfuse Status: True
2024-05-06 01:01:09.616: Bootloader Security Authentication Status: False
2024-05-06 01:01:09.616: Bootloader Security Rdc Status: False
2024-05-06 01:01:09.617: Effective Bootloader Security Status: True
2024-05-06 01:01:09.617: Native Debug Status: False
2024-05-06 01:01:09.889: Phone type: RM-1045
2024-05-06 01:01:09.889: Product code: 059W0X6
2024-05-06 01:01:09.889: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-06 01:01:09.889: Firmware version: 02540.00019.15236.45000
2024-05-06 01:01:09.889: IMEI: 353633062332557
2024-05-06 01:01:09.889: Flash app: 2.59
2024-05-06 01:01:09.889: Flash protocol: 2.34
2024-05-06 01:01:09.889: SecureBoot: Enabled
2024-05-06 01:01:09.889: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-06 01:01:09.889: JTAG: Disabled
2024-05-06 01:01:09.890: Bootloader: Lumia Bootloader Spec B
2024-05-06 01:01:09.890: ProductCode: 059W0X6
2024-05-06 01:01:09.890: ProductType: RM-1045
2024-05-06 01:05:24.022: Initializing flash...
2024-05-06 01:05:24.050: Flashing original FFU...
2024-05-06 01:07:42.517: Rebooting phone
2024-05-06 01:07:42.518: Flash successful!
2024-05-06 01:07:42.596: Lumia disconnected
2024-05-06 01:07:46.669: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 01:07:46.669: Device path: \\?\USB#VID_0421&PID_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 01:07:46.669: Connected device: Lumia
2024-05-06 01:07:46.669: Mode: Bootloader
2024-05-06 01:07:55.654: Lumia disconnected
2024-05-06 01:08:29.926: Found device on interface: 08324f9c-b621-435c-859b-ae4652481b7c
2024-05-06 01:08:29.926: Device path: \\?\USB#VID_0421&PID_06FC&MI_03#6&3aa93d4e&0&0003#{08324f9c-b621-435c-859b-ae4652481b7c}
2024-05-06 01:08:29.926: Connected device: Lumia
2024-05-06 01:08:29.926: Mode: Normal
2024-05-06 01:08:30.613: Operator: 000-33
2024-05-06 01:08:30.616: Manufacturer Model Name: RM-1045_1006
2024-05-06 01:08:30.628: Product Code: 059W0X6
2024-05-06 01:08:30.629: Firmware: 02540.00019.15236.45005
2024-05-06 01:08:30.632: IMEI: 353633062332557
2024-05-06 01:08:30.645: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 01:08:30.648: Bluetooth MAC: 10 2F 6B BD 91 12
2024-05-06 01:08:30.657: WLAN MAC: 10 2F 6B BF E5 68
2024-05-06 01:08:30.667: Bootloader Security: Enabled
2024-05-06 01:08:30.678: Simlock: Unlocked
2024-05-06 01:10:19.781: Start unlock. Phone needs to switch to Flash-mode
2024-05-06 01:10:19.781: In order to start unlocking the bootloader, the phone needs to be switched to Flash-mode.
2024-05-06 01:10:21.746: Switching to Flash-mode
2024-05-06 01:10:21.801: Rebooting phone to Flash mode
2024-05-06 01:10:21.811: Rebooting phone to Flash mode...
2024-05-06 01:10:24.184: Lumia disconnected
2024-05-06 01:10:29.088: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 01:10:29.088: Device path: \\?\USB#VID_0421&PID_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 01:10:29.088: Connected device: Lumia
2024-05-06 01:10:29.088: Mode: Flash
2024-05-06 01:10:29.360: Phone type: RM-1045
2024-05-06 01:10:29.360: Product code: 059W0X6
2024-05-06 01:10:29.360: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-06 01:10:29.360: Firmware version: 02540.00019.15236.45000
2024-05-06 01:10:29.360: IMEI: 353633062332557
2024-05-06 01:10:29.360: Flash app: 2.59
2024-05-06 01:10:29.360: Flash protocol: 2.34
2024-05-06 01:10:29.360: SecureBoot: Enabled
2024-05-06 01:10:29.360: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-06 01:10:29.360: JTAG: Disabled
2024-05-06 01:11:41.082: FFU not added, because it was already present in the repository.
2024-05-06 01:11:41.083: Adding emergency files to repository: C:\edp.edp
2024-05-06 01:11:41.083: Type: RM-1045
2024-05-06 01:11:41.098: Error: Indeks wykracza poza granice tablicy.
2024-05-06 01:36:52.864: Windows Phone Internals version 2.8.6950.39616
2024-05-06 01:36:52.866: Copyright Heathcliff74 / wpinternals.net
2024-05-06 01:36:53.050: Found device on interface: 08324f9c-b621-435c-859b-ae4652481b7c
2024-05-06 01:36:53.050: Device path: \\?\usb#vid_0421&pid_06fc&mi_03#6&3aa93d4e&0&0003#{08324f9c-b621-435c-859b-ae4652481b7c}
2024-05-06 01:36:53.050: Connected device: Lumia
2024-05-06 01:36:53.050: Mode: Normal
2024-05-06 01:36:53.837: Operator: 000-33
2024-05-06 01:36:53.841: Manufacturer Model Name: RM-1045_1006
2024-05-06 01:36:53.891: Product Code: 059W0X6
2024-05-06 01:36:53.894: Firmware: 02540.00019.15236.45005
2024-05-06 01:36:53.915: IMEI: 353633062332557
2024-05-06 01:36:53.939: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 01:36:53.948: Bluetooth MAC: 10 2F 6B BD 91 12
2024-05-06 01:36:53.975: WLAN MAC: 10 2F 6B BF E5 68
2024-05-06 01:36:54.004: Bootloader Security: Enabled
2024-05-06 01:36:54.033: Simlock: Unlocked
2024-05-06 01:36:56.992: Start unlock. Phone needs to switch to Flash-mode
2024-05-06 01:36:56.992: In order to start unlocking the bootloader, the phone needs to be switched to Flash-mode.
2024-05-06 01:37:00.091: Switching to Flash-mode
2024-05-06 01:37:00.137: Rebooting phone to Flash mode
2024-05-06 01:37:00.152: Rebooting phone to Flash mode...
2024-05-06 01:37:02.100: Lumia disconnected
2024-05-06 01:37:07.010: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 01:37:07.010: Device path: \\?\USB#VID_0421&PID_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 01:37:07.010: Connected device: Lumia
2024-05-06 01:37:07.011: Mode: Flash
2024-05-06 01:37:07.331: Phone type: RM-1045
2024-05-06 01:37:07.331: Product code: 059W0X6
2024-05-06 01:37:07.331: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-06 01:37:07.331: Firmware version: 02540.00019.15236.45000
2024-05-06 01:37:07.331: IMEI: 353633062332557
2024-05-06 01:37:07.331: Flash app: 2.59
2024-05-06 01:37:07.331: Flash protocol: 2.34
2024-05-06 01:37:07.331: SecureBoot: Enabled
2024-05-06 01:37:07.331: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-06 01:37:07.331: JTAG: Disabled
2024-05-06 01:40:35.866: FFU not added, because it was already present in the repository.
2024-05-06 01:41:25.820: FFU not added, because it was already present in the repository.
2024-05-06 01:41:25.822: Adding emergency files to repository: C:\ede.ede
2024-05-06 01:41:25.822: Type: RM-1045
2024-05-06 01:41:25.842: Error: Indeks wykracza poza granice tablicy.
2024-05-06 01:52:14.570: Windows Phone Internals version 2.8.6950.39616
2024-05-06 01:52:14.571: Copyright Heathcliff74 / wpinternals.net
2024-05-06 01:52:14.798: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 01:52:14.798: Device path: \\?\usb#vid_0421&pid_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 01:52:14.798: Connected device: Lumia
2024-05-06 01:52:14.798: Mode: Flash
2024-05-06 01:52:14.925: Security flags: 0x00007BFF
2024-05-06 01:52:14.950: Platform Name: Nokia.MSM8974.P6180.2.2
2024-05-06 01:52:14.977: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 01:52:15.001: Root Key Hash: 80 0E EB 50 8F 7B BD F1 2A 19 26 26 21 FD 83 72 97 A3 B0 62 FE 2A 70 78 C0 F3 16 7E 57 F2 12 17
2024-05-06 01:52:15.001: Platform Secure Boot Status: True
2024-05-06 01:52:15.002: Uefi Secure Boot Status: True
2024-05-06 01:52:15.003: Effective Secure Boot Status: True
2024-05-06 01:52:15.003: Bootloader Security Qfuse Status: True
2024-05-06 01:52:15.004: Bootloader Security Authentication Status: False
2024-05-06 01:52:15.005: Bootloader Security Rdc Status: False
2024-05-06 01:52:15.006: Effective Bootloader Security Status: True
2024-05-06 01:52:15.007: Native Debug Status: False
2024-05-06 01:52:15.292: Phone type: RM-1045
2024-05-06 01:52:15.292: Product code: 059W0X6
2024-05-06 01:52:15.292: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-06 01:52:15.292: Firmware version: 02540.00019.15236.45000
2024-05-06 01:52:15.292: IMEI: 353633062332557
2024-05-06 01:52:15.293: Flash app: 2.59
2024-05-06 01:52:15.293: Flash protocol: 2.34
2024-05-06 01:52:15.293: SecureBoot: Enabled
2024-05-06 01:52:15.293: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-06 01:52:15.293: JTAG: Disabled
2024-05-06 01:52:15.293: Bootloader: Lumia Bootloader Spec B
2024-05-06 01:52:15.293: ProductCode: 059W0X6
2024-05-06 01:52:15.294: ProductType: RM-1045
2024-05-06 01:53:15.963: Error: Bad FFU file
2024-05-06 01:53:15.963: Bad FFU file: C:\ede.ede.
Nieprawidłowy format pliku wykonywalnego (.exe) lub biblioteki (.dll).
2024-05-06 01:53:15.963:     Error: Nieprawidowy format pliku wykonywalnego (.exe) lub biblioteki (.dll).
2024-05-06 01:53:32.913: Error: Bad FFU file
2024-05-06 01:53:32.913: Bad FFU file: C:\edp.edp.
Nieprawidłowy format pliku wykonywalnego (.exe) lub biblioteki (.dll).
2024-05-06 01:53:32.913:     Error: Nieprawidowy format pliku wykonywalnego (.exe) lub biblioteki (.dll).
2024-05-06 02:02:38.656: FFU not added, because it was already present in the repository.
2024-05-06 02:02:38.657: Adding emergency files to repository: C:\edp.edp
2024-05-06 02:02:38.658: Type: RM-1045
2024-05-06 02:02:38.661: Error: Indeks wykracza poza granice tablicy.
2024-05-06 02:10:15.089: Windows Phone Internals version 2.8.6950.39616
2024-05-06 02:10:15.091: Copyright Heathcliff74 / wpinternals.net
2024-05-06 02:10:15.258: Found device on interface: 08324f9c-b621-435c-859b-ae4652481b7c
2024-05-06 02:10:15.258: Device path: \\?\usb#vid_0421&pid_06fc&mi_03#6&3aa93d4e&0&0003#{08324f9c-b621-435c-859b-ae4652481b7c}
2024-05-06 02:10:15.258: Connected device: Lumia
2024-05-06 02:10:15.258: Mode: Normal
2024-05-06 02:10:15.991: Operator: 000-33
2024-05-06 02:10:15.994: Manufacturer Model Name: RM-1045_1006
2024-05-06 02:10:16.037: Product Code: 059W0X6
2024-05-06 02:10:16.039: Firmware: 02540.00019.15236.45005
2024-05-06 02:10:16.055: IMEI: 353633062332557
2024-05-06 02:10:16.068: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 02:10:16.072: Bluetooth MAC: 10 2F 6B BD 91 12
2024-05-06 02:10:16.083: WLAN MAC: 10 2F 6B BF E5 68
2024-05-06 02:10:16.097: Bootloader Security: Enabled
2024-05-06 02:10:16.112: Simlock: Unlocked
2024-05-06 02:10:19.294: Start unlock. Phone needs to switch to Flash-mode
2024-05-06 02:10:19.294: In order to start unlocking the bootloader, the phone needs to be switched to Flash-mode.
2024-05-06 02:10:22.133: Switching to Flash-mode
2024-05-06 02:10:22.172: Rebooting phone to Flash mode
2024-05-06 02:10:22.182: Rebooting phone to Flash mode...
2024-05-06 02:10:24.012: Lumia disconnected
2024-05-06 02:10:28.939: Found device on interface: 9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc
2024-05-06 02:10:28.939: Device path: \\?\USB#VID_0421&PID_0714#5&fa9ae3&0&4#{9e3bd5f7-9690-4fcc-8810-3e2650cd6ecc}
2024-05-06 02:10:28.939: Connected device: Lumia
2024-05-06 02:10:28.939: Mode: Flash
2024-05-06 02:10:29.227: Phone type: RM-1045
2024-05-06 02:10:29.227: Product code: 059W0X6
2024-05-06 02:10:29.227: Root key hash: 800EEB508F7BBDF12A19262621FD837297A3B062FE2A7078C0F3167E57F21217
2024-05-06 02:10:29.227: Firmware version: 02540.00019.15236.45000
2024-05-06 02:10:29.227: IMEI: 353633062332557
2024-05-06 02:10:29.227: Flash app: 2.59
2024-05-06 02:10:29.227: Flash protocol: 2.34
2024-05-06 02:10:29.227: SecureBoot: Enabled
2024-05-06 02:10:29.227: Flash app security: Enabled (FFU security: Enabled, RDC: Not found, Authenticated: False)
2024-05-06 02:10:29.227: JTAG: Disabled
2024-05-06 02:11:43.426: FFU not added, because it was already present in the repository.
2024-05-06 02:11:43.428: Adding emergency files to repository: C:\edp.edp
2024-05-06 02:11:43.428: Type: RM-1045
2024-05-06 02:11:43.438: Error: Indeks wykracza poza granice tablicy.
2024-05-06 02:36:33.937: Windows Phone Internals version 2.8.6950.39616
2024-05-06 02:36:33.939: Copyright Heathcliff74 / wpinternals.net
2024-05-06 02:36:34.119: Found device on interface: 08324f9c-b621-435c-859b-ae4652481b7c
2024-05-06 02:36:34.119: Device path: \\?\usb#vid_0421&pid_06fc&mi_03#6&3aa93d4e&0&0003#{08324f9c-b621-435c-859b-ae4652481b7c}
2024-05-06 02:36:34.119: Connected device: Lumia
2024-05-06 02:36:34.119: Mode: Normal
2024-05-06 02:36:34.829: Operator: 000-33
2024-05-06 02:36:34.850: Manufacturer Model Name: RM-1045_1006
2024-05-06 02:36:34.906: Product Code: 059W0X6
2024-05-06 02:36:34.909: Firmware: 02540.00019.15236.45005
2024-05-06 02:36:34.925: IMEI: 353633062332557
2024-05-06 02:36:34.948: Public ID: 6C F2 C3 70 7E BB 00 9A 33 4F 9E EE AE 7F A9 31 A1 A4 35 23
2024-05-06 02:36:34.969: Bluetooth MAC: 10 2F 6B BD 91 12
2024-05-06 02:36:35.015: WLAN MAC: 10 2F 6B BF E5 68
2024-05-06 02:36:35.045: Bootloader Security: Enabled
2024-05-06 02:36:35.082: Simlock: Unlocked
2024-05-06 15:18:36.547: Lumia disconnected

**Comment by gus33000** [comment]
It appears you are selecting the ede in the "add ffu to repo" feature. This feature can only add ffu files. Not emergency files. You can only specify an ede file (and not an edp file) in the unlock screen where it asks for files in the field to select an emergency programmer. Be aware you may have multiple feeds. One for your ffu image. One for the ede, and one optionally for an ffu with a supported os version as a donnor.

**Comment by MackaG** [comment]
It was thing i were trying to do to make tool work. Thats why I included only "important part" of log. WPInternals were doing same thing, outputing same error on a clean Windows 7 install without doing anything. Only thing I did was connect lumia 930 to PC, enter unlock bootloader section and select files that were needed to unlock bootloader, and it crashed. I can provide you log from that windows 7 PC if you ask

**Comment by MackaG** [comment]
Accidentally selected close with comment

**Comment by MackaG** [comment]
hello?

**Comment by MackaG** [comment]
OK, I figured it out. Lumiafirmware.com's ftp protocol is broken and it inserts fragment of HTML script before actual data. When I removed it with HxD it moved further, but there's another error. Good old failed to write to pipe, and reported exception ".".(?). I dont know but I'm happy that at least it partially works. Also, I can use FFU file as emergency programmer for some reason and it gives me same error.
